components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -173,9 +185,13 @@ impl ScanExecutorImpl for IndexScanExecutorImpl {
         columns: &mut LazyBatchColumnVec,
     ) -> Result<()> {
         if value.len() > MAX_OLD_ENCODED_VALUE_LEN {
-            self.process_kv_pair_new(key, value, columns)
+            if value[0] <= 1 && value[1] == table::COMMON_HANDLE_FLAG {
+                self.process_unique_common_handle_value(key, value, columns)
+            } else {
+                self.process_normal_new_collation_value(key, value, columns)
+            }
         } else {
-            self.process_kv_pair_old(key, value, columns)
+            self.process_normal_old_collation_value(key, value, columns)
```

Comment:
Why the process of the key (common handle / int handle) is determined by the value? Can we ensure that for clustered index, the value must be `> MAX_OLD_ENCODED_VALUE_LEN`?

Reply:
We can ensure it.

Reply:
Can we check it according to the `DecodeHandleStrategy` proposed in https://github.com/tikv/tikv/pull/8179/files#r448795587, which looks to be more general, and no need to check against values. (precondition: a clustered index table will not mix int handle / common handle).

Reply:
Sure we could, however, if the request requires to fetch values that some are in the new format and the other in the old format then the decoding strategy might fail.

Reply:
I think "if the request requires to fetch values that some are in the new format and the other in the old format then the decoding strategy might fail" will never happen since a table will not mix two different formats. Additionally, if it happens, the code will panic elsewhere, since the ColumnVec container is created according to `DecodeHandleStrategy` and pushing two different formats into it will lead to panic.

Reply:
For old normal collation index value, we may have common handles or int handles in our keys or values. We need to check the key and accords the results to extract the rest of `key/value` in common handle way or int handle way. In other words, we could not get enough information while we know what strategy is used to decode the handle. We need to know more about key/value's length.

If we check the kV pairs according to `DecodeHandleStrategy`, it still needs to check against the value due to we need to know what kind of collation the int handle is hiding, it may be in a new collation or old collation format.

Reply:
Yes, I mean, for common handle / int handle, need to check against `DecodeHandleStrategy`. `DecodeHandleStrategy` does not contain any information about the collation so you cannot check collation format by looking at `DecodeHandleStrategy`. If you do not check against `DecodeHandleStrategy`, when there is indeed mixed data, TiKV will panic. The problem is it will cause TiKV panic due to unmatched container (e.g. pushing raw value into int container) instead of checking value or not.